homeassistant/components/amcrest/camera.py

```diff
@@ -90,6 +92,10 @@
     async_add_entities([AmcrestCam(name, device, hass.data[DATA_FFMPEG])], True)
 
 
+class _CannotSnapshot(Exception):
```

Comment:
Why underscore this? The platform won't share things anyway.

Reply:
I thought for the usual reason. To mark it as not intended for use outside the module. But I can delete the underscore if you think it would be better that way.

Reply:
Yes, nothing from the platform should be shared, so we don't need to mark it.

Reply:
Ok, I'll change it. I was just trying to follow PEP 8.

Reply:
BTW, not to be argumentative, but platforms _do_ share; e.g., SCAN_INTERVAL, setup, async_setup, etc. And there is precedence for using an underscore for an exception class only intended for use within a module. E.g., see `_StopScript` & `_SuspendScript` in `homeassistant.helpers.script`.

I'll still remove the underscore if you want, but I don't think that's right for several reasons.

Reply:
All interfaces in the platform that are not defined by our core architecture should by default be considered internal.

Reply:
Yes, and PEP8 says that all internal module level names should start with an underscore. I guess I don't understand why you're insisting that I don't follow PEP8, when PEP8 is like the first thing mentioned in the [Style guidelines](https://developers.home-assistant.io/docs/en/development_guidelines.html), but I guess it's not worth discussing anymore. I'll change it.

Reply:
This is from PEP8:

> An interface is also considered internal if any containing namespace (package, module or class) is considered internal.

Reply:
Ah, my point exactly. The module this class is defined in is _not_ "internal", at least not in the sense that PEP8 means. See the previous paragraph.

>internal interfaces (packages, modules, classes, functions, attributes or other names) should still be prefixed with a single leading underscore. Platform modules are not named with leading underscores, precisely because they are _not_ "internal."

The platform module is not internal to the integration. It is not named with a leading underscore. It is public because it is intended to be used by other parts of HA. But the class defined inside this module _is_ internal, because it is not intended to be used by other parts of HA.

Like I said, I'll remove it if you insist, but I still disagree with your reasoning.

Reply:
The platform is considered internal except for the core interfaces.